src/pages/workspaces/workspace/useWorkspace.ts

```diff
@@ -174,73 +175,86 @@ export const useWorkspace = (namespace, name): WorkspaceDetails => {
-      } = workspace;
+    const {
+      accessLevel,
+      workspace: { createdBy, createdDate, googleProject },
+    } = workspace;
 
-      checkWorkspaceInitialization(workspace);
+    checkWorkspaceInitialization(workspace);
 
-      // Request a service account token. If this is the first time, it could take some time before everything is in sync.
-      // Doing this now, even though we don't explicitly need it now, increases the likelihood that it will be ready when it is needed.
-      if (canWrite(accessLevel) && isGoogleWorkspace(workspace)) {
-        saToken(googleProject);
-      }
+    // Request a service account token. If this is the first time, it could take some time before everything is in sync.
+    // Doing this now, even though we don't explicitly need it now, increases the likelihood that it will be ready when it is needed.
+    if (canWrite(accessLevel) && isGoogleWorkspace(workspace)) {
+      saToken(googleProject);
+    }
 
-      // This is old code-- it is unclear if this case can actually happen anymore.
-      if (!isOwner(accessLevel) && createdBy === getTerraUser().email && differenceFromNowInSeconds(createdDate) < 60) {
-        accessNotificationId.current = notify('info', 'Workspace access synchronizing', {
-          message: h(Fragment, [
-            'It looks like you just created this workspace. It may take up to a minute before you have access to modify it. Refresh at any time to re-check.',
-            div({ style: { marginTop: '1rem' } }, [
-              h(
-                Link,
-                {
-                  onClick: () => {
-                    refreshWorkspace();
-                    clearNotification(accessNotificationId.current);
-                  },
+    // This is old code-- it is unclear if this case can actually happen anymore.
+    if (!isOwner(accessLevel) && createdBy === getTerraUser().email && differenceFromNowInSeconds(createdDate) < 60) {
+      accessNotificationId.current = notify('info', 'Workspace access synchronizing', {
+        message: h(Fragment, [
+          'It looks like you just created this workspace. It may take up to a minute before you have access to modify it. Refresh at any time to re-check.',
+          div({ style: { marginTop: '1rem' } }, [
+            h(
+              Link,
+              {
+                onClick: () => {
+                  refreshWorkspace();
+                  clearNotification(accessNotificationId.current);
                 },
-                ['Click to refresh now']
-              ),
-            ]),
+              },
+              ['Click to refresh now']
+            ),
           ]),
-        });
-      }
-    } catch (error: any) {
-      if (error.status === 404) {
-        setAccessError(true);
-      } else {
-        throw error;
-      }
+        ]),
+      });
     }
-  }) as () => Promise<void>;
+  };
+
+  const checkForAccessError: ErrorCallback = (error: unknown) => {
```

Comment:
This needed to be broken out separately from the refresh functionality, so that we could allow custom error handling when refreshing silently.